FirebaseCore/Extension/FIRHeartbeatLogger.h

```diff
@@ -35,13 +35,18 @@ typedef NS_ENUM(NSInteger, FIRDailyHeartbeatCode) {
 /// Asynchronously logs a heartbeat.
 - (void)log;
 
+/// Gets the heartbeat code for today.
+- (FIRDailyHeartbeatCode)heartbeatCodeForToday;
+
 #ifndef FIREBASE_BUILD_CMAKE
-/// Return the headerValue for the HeartbeatLogger.
+/// Return the header value for the heartbeat logger.
 - (NSString *_Nullable)headerValue;
-#endif  // FIREBASE_BUILD_CMAKE
 
-/// Gets the heartbeat code for today.
-- (FIRDailyHeartbeatCode)heartbeatCodeForToday;
+@optional
```

Comment:
Marking this optional to prevent having to change the FakeHeartbeatLogger implementations across all of the test suites that make a fake logger (fcm, firestore, fis, etc.). This is why a force unwrap is necessary on line 103 (new diff) of FirebaseAuth/Sources/Swift/Backend/AuthBackend.swift

Reply:
I'd feel better about making `headerValue` optional since we want to encourage future clients to use the async API.

Changing three or four tests seems a small cost to avoid a force unwrap in the library. But I could be convinced otherwise.

Reply:
True. It was simpler than I thought. Instead if marking `headerValue` as optional, I marked it as `NS_SWIFT_UNAVAILABLE`.